packages/react-reconciler/src/ReactFiberWorkLoop.new.js

```diff
@@ -1833,11 +1863,35 @@ function resetChildLanes(completedWork: Fiber) {
         newChildLanes,
         mergeLanes(child.lanes, child.childLanes),
       );
+
+      if (!didBailout) {
+        subtreeTag |= child.subtreeTag & ~DidBailout;
+
+        const effectTag = child.effectTag;
+        if ((effectTag & BeforeMutationMask) !== NoEffect) {
+          subtreeTag |= BeforeMutation;
+        }
+        if ((effectTag & MutationMask) !== NoEffect) {
+          subtreeTag |= Mutation;
+        }
+        if ((effectTag & LayoutMask) !== NoEffect) {
+          subtreeTag |= Layout;
+        }
+      }
+
+      if ((child.effectTag & Incomplete) !== NoEffect) {
+        childrenDidNotComplete = true;
+      }
+
       child = child.sibling;
     }
   }
 
   completedWork.childLanes = newChildLanes;
+
+  if (!childrenDidNotComplete) {
```

Comment:
I wonder why this check is necessary. `resetChildLanes` shouldn't be called if something in the subtree didn't complete. So I wouldn't expect `childrenDidNotComplete` to  ever be true.

Do you remember if any tests failed without this guard?

Reply:
I don't remember exactly why this was here (I think left over from a previous implementation when tests were failing to match with what the effect list was doing [here](https://github.com/facebook/react/blob/7171494026b256a5e0778414e3d0bf1d4c0e9dee/packages/react-reconciler/src/ReactFiberWorkLoop.new.js#L1681). All the tests pass without this though so I'll remove.